march_hardware/src/joint.cpp

```diff
@@ -160,57 +148,12 @@ void Joint::actuateTorque(int16_t target_torque)
     throw error::HardwareException(error::ErrorType::NOT_ALLOWED_TO_ACTUATE, "Joint %s is not allowed to actuate",
                                    this->name_.c_str());
   }
-  this->imc_->actuateTorque(target_torque);
+  this->controller_->actuateTorque(target_torque);
 }
 
 int16_t Joint::getTorque()
 {
-  if (!this->hasIMotionCube())
-  {
-    ROS_WARN("[%s] Has no iMotionCube", this->name_.c_str());
-    return -1;
-  }
-  return this->imc_->getTorque();
-}
-
-int32_t Joint::getAngleIUAbsolute()
-{
-  if (!this->hasIMotionCube())
-  {
-    ROS_WARN("[%s] Has no iMotionCube", this->name_.c_str());
-    return -1;
-  }
-  return this->imc_->getAngleIUAbsolute();
-}
-
-int32_t Joint::getAngleIUIncremental()
-{
-  if (!this->hasIMotionCube())
-  {
-    ROS_WARN("[%s] Has no iMotionCube", this->name_.c_str());
-    return -1;
-  }
-  return this->imc_->getAngleIUIncremental();
-}
-
-double Joint::getVelocityIUAbsolute()
-{
-  if (!this->hasIMotionCube())
-  {
-    ROS_WARN("[%s] Has no iMotionCube", this->name_.c_str());
-    return -1;
-  }
-  return this->imc_->getVelocityIUAbsolute();
-}
-
-double Joint::getVelocityIUIncremental()
-{
-  if (!this->hasIMotionCube())
-  {
-    ROS_WARN("[%s] Has no iMotionCube", this->name_.c_str());
-    return -1;
-  }
-  return this->imc_->getVelocityIUIncremental();
+  return this->controller_->getTorque();
```

Comment:
This will end horribly, if the `controller_` is a `nullptr`, always check you pointers before using them. I now see you do this a lot more, so be warned

Reply:
I build a check in hardware builder to ensure that every joint has a controller at startup, removing the need to check this every time during runtime.

Reply:
The constructor would only check this once. Adding a check here would ensure that it is always a valid pointer.